### ISSUE TITLE
Adding Dynamic Data to HTTP Params

### ISSUE THREAD

**Issue by njho** [transistorsoft/react-native-background-geolocation#420]
## Your Environment
* Plugin version: 2.11
* Platform: Android
* OS version: 7.0
* Device manufacturer / model: Samsung Galaxy S8
* React Native version (`react-native -v`):  0.48.4
* Plugin config

## Expected Behavior
No bugs

## Actual Behavior
No bugs

## Steps to Reproduce
No bugs

## Context
How do you add dynamic data to the HTTP Params? 

Is there a way to to add a custom timestamp, or something like that?

## Debug logs
N/A

**Comment by christocracy** [comment]
No.  Why would you need a custom timestamp?  Your server can consume the plugin's [ISO-8601 formatted UTC](https://en.wikipedia.org/wiki/ISO_8601) timestamp and reformat it any way it desires.

By the time the Javascript `location` event has fired, and the location has already been persisted to the database.  If you've configured `autoSync: true`, the HTTP request has already occurred.

**Comment by njho** [comment]
We wouldn't need a custom timestamp. It was just more of an example

But for instance, I couldn't have any dynamic data in the body:  

```
 params: {
    device_id: 'abc123',
    random_num: Math.random()
  }
```

**Comment by christocracy** [comment]
You can always:
```javascript
BackgroundGeolocation.setConfig({
  params: {
    device_id: 'abc123',
    random_num: Math.random()
  }
});
```

But if you're hoping to append these params from the `location` event, that's not going to work

```javascript
BackgroundGeolocation.on('location', (location) => {
  // the supplied location here has already been persisted and HTTP request already executed.

  // These params will be applied to NEXT http request.
  BackgroundGeolocation.setConfig({foo: 'bar'});
 
});
```

**Comment by njho** [comment]
Hey Chris,


I purchased the plugin and posted an issue here<https://github.com/transistorsoft/react-native-background-geolocation/issues/439>. What's the best way to do this? (I also played for plugin in case you're wondering :) )


Context

I am trying to call a function for every location that is unique AND persisted to database. In effect, every location that my server application sees as a real event.

Current problems I'm bumping into:

  1.  Cannot solely rely on Location event: Location events are not ALL posted to HTTP, so I may be collecting and posting extraneous data.
  2.  Cannot solely rely on HTTP event: Should phone be out of network, the HTTP event will still be triggered, but there will be no reference to the UUID, or location it was trying to POST, making it difficult to sync info with my function(uuid)...

Possible Solutions Being Considered

  1.  Collect Information for every location event - which would cause some extra bloat on app, then, upload to database for every successful HTTP event that contains uuid.
  2.  Create some sort of database, that can somehow cross-reference whether a location is also an http event.

Is there some sort of better solution for this? What I'm considering seems to be less than ideal :)

Thanks Christocracy




________________________________
